client/src/main/scala/drt/client/SPAMain.scala

```diff
@@ -39,6 +44,32 @@ object SPAMain {
       }
     }
 
+    def withDateAndTime(date: Option[String], timeStart: Option[String], timeEnd: Option[String]): TerminalPageTabLoc = {
```

Comment:
Trying to think of a neater solution here. Off the top of my head would it work to to use case classes with implementing a 'UrlParameter' trait? Eg

`
sealed trait UrlParameter {
   val name: String
   val value: Option[String]
}

case class UrlDate(value: Option[String]) extends UrlParameter {
   val name: String = "date"
}
`

You could then pass a list of UrlParameters which are leftFolded to either set or unset query parameters?

Reply:
done